### DISCUSSION TITLE
How can I convert my `ltex-ls` lua Neovim config to toml for Helix?

### DISCUSSION THREAD

**Discussion by David-Else** [helix-editor/helix#5450] [Q&A] [answered]
In Neovim, this lua works with https://github.com/valentjn/ltex-ls:

```
  settings = {
    ltex = {
      disabledRules = {
        ['en-US'] = { 'PROFANITY' },
        ['en-GB'] = { 'PROFANITY' },
      },
      dictionary = {
        ['en-US'] = [ "madeupword", "anothermadeupword" ],
        ['en-GB'] = [ "madeupword", "anothermadeupword" ],
      },
    },
  },
```
All the settings are show here: https://valentjn.github.io/ltex/settings.html

In Helix the config format is:

```
[language.config]
cargo = { allFeatures = true }
checkOnSave = { command = "clippy" }
```
How do I convert this to work in Helix? Thanks!

**Comment by David-Else** [comment, marked as answer]
I did it! :)

You should be able to switch languages using front matter in Markdown, and settings must be for each language, hence I have them twice for US and GB English.
```
config = { ltex.disabledRules = { "en-US" = [
  "PROFANITY",
], "en-GB" = [
  "PROFANITY",
] }, ltex.dictionary = { "en-US" = [
  "madeupword",
], "en-GB" = [
  "madeupword",
] } }
```

**Reply by David-Else** [reply]
Follow up question, why can't I load a spelling file like in Neovim with `"en-US" = [
  ":/home/user/spelling.txt" ]` ?

**Reply by Marsevil** [reply]
Hi, I'm experimenting the same issue, do you have any news about the loading of an external dictionary file ?

**Reply by nshern** [reply]
Hi, I am similarly experimenting with ltex-ls, have you guys figured this out?

**Reply by cunha** [reply]
The reason why it doesn't work is that the loading of external files (starting with `:`) is handled in the VSCode extension; there likely isn't code in LTeX-LS or LTeX to handle this. References:

https://github.com/valentjn/vscode-ltex/blob/74484ff6e0f1f810fc24d57ba51286c347b31cf3/src/ExternalFileManager.ts#L255-L267

https://github.com/valentjn/vscode-ltex/blob/74484ff6e0f1f810fc24d57ba51286c347b31cf3/src/ExternalFileManager.ts#L360-L362

I think the proper fix is to implement file handling in LTeX-LS or LTeX itself, where it would work for all LSP clients.
